Makefile

```diff
@@ -424,6 +425,9 @@ endif
 ifdef WINDOWS_CODESIGN_PFX_PATH
 	scripts/sign-windows-bin.sh $(INSTALLER_OUTPUT)
 endif
+	echo -e $(BUILD_MSG) "7z"
+	rm $(OUTPUT)/status.iss
+	7z a $(STATUS_CLIENT_7Z) ./$(OUTPUT)
```

Comment:
Why isn't this a separate target tho?

Reply:
We just keep making the `$(STATUS_CLIENT_EXE)` target bigger and bigger, this really needs refactoring.

Reply:
As per the ticket #3324 we need to release both the installer and the archive at the same time.
If we make them separate, then we will drastically increase the build time, because compiling, linking and deployqt takes the biggest half of the overall build time.
However, for testing/QA we probably need to be able to skip the achieving step. I welcome any suggestions on this.

Reply:
Yes, but you can just make it a separate makefile target without compiling things multiple times.

Reply:
Look, here's a simple example:
```Makefile
binary:
	touch app.bin

app.exe: binary
	cp app.bin app.exe

app.zip: binary
	cp app.bin app.zip
```
```
 > make binary
touch app.bin

 > rm app.*
 > make app.exe
touch app.bin
cp app.bin app.exe

 > rm app.*
 > make app.exe app.zip
touch app.bin
cp app.bin app.exe
cp app.bin app.zip
```
See?

Reply:
I see! Will try changing as suggested.

Reply:
But in the end, in Jenkins we will have to execute both targets anyways?
p.s. I understand we still need to separate targets, no objections here.

Reply:
Alright, I have done the first round, introduced the new target `zip-windows` (similar to tgz-linux) and this seems working.
However, when I tested this locally, I noticed that running `pkg-windows` and `zip-windows` start all new build with windeployqt with creating inno installer, etc. Even though `zip-windows` has a dependency on `pkg-windows` output.
Also, not related to this, but in general I found it very annoying that `make` always re-build Qt resources, always recompile nim application's source code, etc. It does not rebuild deps like status-go, but the whole app itself is being rebuilt all the time. I think this where I need help. Or at least, to understand why `zip-windows` triggers whole app rebuild  even though it depends on the installer produced on the previous step (pkg/Status.exe).

Reply:
> I found it very annoying that make always re-build Qt resources, always recompile nim application's source code, etc. It does not rebuild deps like status-go, but the whole app itself is being rebuilt all the time. I think this where I need help.

I'll be happy to try to help.

The reason it's this way, historically, is that we weren't sure how to setup the dependencies logic for some `make` targets (e.g. default PHONY target, which equates to `nim_status_client:`) so that when a `.nim` or `.qml` source is changed that e.g. `make V=1` results in a rebuild but when none has changed it does not rebuild.

We could simply have a dependency for `nim_status_client:` be `bin/nim_status_client`. But that would mean that if you want/need a rebuild re: changed `.nim` sources you'd need to `rm bin/nim_status_client` before running `make`. It's the same situation for the `rcc:` target and the `.qml` sources.

There is probably a good way to deal with this, need to research it a bit.

Reply:
> why `zip-windows` triggers whole app rebuild even though it depends on the installer produced on the previous step (pkg/Status.exe).

In the Makefile in this PR branch:

`zip-windows:` depends on `$(STATUS_CLIENT_7Z)`.
`$(STATUS_CLIENT_7Z)` depends on `$(STATUS_CLIENT_EXE)`.
`$(STATUS_CLIENT_EXE)` depends on `nim_status_client`.
`nim_status_client` does not have a dependency related to artifacts that it produces; it also depends on `rcc`, which likewise does not have a dependency related to artifacts it produces. For that reason, there is always a re/build when those targets are run.

`$(STATUS_CLIENT_7Z)` should probably depend on target that forms a common subset for `$(STATUS_CLIENT_EXE)` and ``$(STATUS_CLIENT_7Z)`, i.e. so that it doesn't unnecessarily build the `.exe` installer.

Reply:
>We could simply have a dependency for `nim_status_client`: be `bin/nim_status_client`. But that would mean that if you want/need a rebuild for changed `.nim` sources you'd need to `rm bin/nim_status_client` before running make. It's the same situation for the `rcc:` target and the `.qml` sources.

@michaelsbradleyjr This is simply not true. This is a fundamental misunderstanding of how `make` works.
You can simply specify the source files as dependencies for a target:
```Makefile
app.bin: src/dir/*
app.bin: src/*
	touch app.bin
```
```
 > make app.bin
touch app.bin

 > make app.bin
make: 'app.bin' is up to date.

 > touch src/a.c 
 > make app.bin 
touch app.bin

 > make app.bin
make: 'app.bin' is up to date.

 > touch src/dir/x.c 
 > make app.bin     
touch app.bin
```
And a dependency can also be generated via a command, like:
```Makefile
app.bin: $(shell find src -name '*.c')
	touch app.bin
```

Reply:
Normally you'd put it in a variable tho:
```Makefile
NIM_SOURCES := $(shell find src -name '*.nim')

bin/app: $(NIM_SOURCES)
	nim c ...
```

Reply:
> This is just not true. This is a fundamental misunderstanding of how make works

I meant if you simply did it without setting up the proper additional dependency/ies.

Way back when, those of us on the desktop team didn't know what we're doing with `Makefile`, so the `bin/nim_status_client` dependency was introduced naively. Then someone got annoyed that they had to delete e.g. `bin/nim_status_client` all the time so it was removed and we've lived with the "always rebuild" behavior until now.

I will try implementing what you suggested, thanks for the tips (still lots I don't know about `make`), seems like it will be a nice improvement.

Reply:
Ping me if you need help. I've worked a fair bit with Makefiles.